perfkitbenchmarker/providers/openstack/os_virtual_machine.py

```diff
@@ -54,21 +55,47 @@ def __init__(self, vm_spec):
                                                FLAGS.run_uri)
         self.client = os_utils.NovaClient()
         self.public_network = os_network.OpenStackPublicNetwork(
-            FLAGS.openstack_public_network
+            FLAGS.openstack_floating_ip_pool
         )
         self.id = None
         self.pk = None
         self.user_name = FLAGS.openstack_image_username
         self.boot_wait_time = None
         self.image = self.image or self.DEFAULT_IMAGE
+        self.public_net = None
+        self.private_net = None
+        self.floating_ip = None
 
     def _Create(self):
         image = self.client.images.findall(name=self.image)[0]
         flavor = self.client.flavors.findall(name=self.machine_type)[0]
 
-        network = self.client.networks.find(
-            label=FLAGS.openstack_private_network)
-        nics = [{'net-id': network.id}]
+        # FIXME(meteofox): Remove --openstack_public_network and
+        # --openstack_private_network once depreciation time has expired
+        self.floating_ip_pool_name = (FLAGS.openstack_floating_ip_pool or
+                                      FLAGS.openstack_public_network)
+        self.network_name = (FLAGS.openstack_network or
+                             FLAGS.openstack_private_network)
+
+        if self.network_name:
+            self.private_net = self.client.networks.find(
+                label=self.network_name)
+        if self.floating_ip_pool_name:
+            self.public_net = self.client.networks.find(
+                label=self.floating_ip_pool_name)
+
+        if self.public_net is None and self.private_net is None:
+            raise errors.Error(
+                'Cannot build instance without a network. Make sure to set '
+                'either just --openstack_network or both --openstack_network '
+                'and --openstack_private_network flags.')
```

Comment:
`--openstack_network` is essentially a rename of  `--openstack_private_network`.  They behave the same way  as before, but we decided to drop 'private' from the name since 'private' is relative to how OpenStack was configured and it caused confusion. For the moment I deprecated `--openstack_private_network` to give some time before I completely remove it.

So, in the case that both are set then we prefer the new one, but we could have chosen any of them, as long as they were the same.

They 'new' behavior is that setting a floating IP pool is optional, and if one is given then whatever network that floating IP pool belongs to, it **must** route to the network set with `--openstack_network` or OpenStack will not allocate a floating IP for the instance since it has no way to reach it.

The usual network configuration for OpenStack[1] is to use an internal network that provides the fixed IPs at build time to the instances and then later a user can attach a floating IP from a public network since usually there is a limited number of IP addresses available for public addressing. 

[1] http://docs.openstack.org/openstack-ops/content/network_design.html#public_addressing